### PR TITLE
Bump aws-sdk-go version to v1.44.145

### PR DESCRIPTION
**Pull request by xdu31**
Bump aws-sdk-go version to v1.44.145

### CODE REVIEW

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
+	//nolint
```
Comment (wongma7) [inline pkg/plugin/plugin.go:24]: why is this needed, do you have the golangci-lint error handy? I don't see the lint error https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_aws-encryption-provider/89/pull-aws-encryption-provider-verify/1598078085045948416

Comment (xdu31) [reply pkg/plugin/plugin.go:24]: > why is this needed, do you have the golangci-lint error handy? I don't see the lint error https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_aws-encryption-provider/89/pull-aws-encryption-provider-verify/1598078085045948416

The `golangci-lint` is complaining about `awserr` not used, which is actually used, there's similar thread about wrong judgement from `golangci-lint`, https://github.com/golangci/golangci-lint/discussions/2214, someone claimed to fix some failure with golang version bump from 1.15 to 1.16, however, `aws-encryption-provider` has a hard limit on golang 1.15 as the highest version due to protobuf.

File: go.mod [hunk 2]
```diff
@@ -31,3 +31,5 @@ replace github.com/golangci/lint-1 v0.0.0-20180610141402-ee948d087217 => github.
 replace github.com/timakin/bodyclose => github.com/golangci/bodyclose v0.0.0-20190714144026-65da19158fa2
 
 replace mvdan.cc/unparam v0.0.0-20190124213536-fbb59629db34 => mvdan.cc/unparam v0.0.0-20190209190245-fbb59629db34
+
+replace golang.org/x/sys => golang.org/x/sys v0.0.0-20220811171246-fbc7d0a398ab
```
Comment (micahhausler) [inline go.mod:35]: Why this change?

Comment (xdu31) [reply go.mod:35]: > Why this change?

@micahhausler This to fix `undefined: unsafe.Slice` errors introduced by the `aws-sdk-go` version bump up. The `unsafe.Slice` was added as part of Go 1.17, we are using Go 1.15 in `aws-encryption-provider`. I took the accept answer in https://stackoverflow.com/questions/74049351/gcloud-functions-deploy-go-runtime-error-undefined-unsafe-slice-error-id-2f5

Comment (micahhausler) [reply go.mod:35]: Go 1.15 is unmaintained... why don't we bump the Go version to something newer. I think you can just bump it in this PR
